api/api.proto

```diff
@@ -3454,6 +3531,110 @@ message SdkRoleUpdateResponse {
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem trim Start request
+  string status = 2;
+}
+
+// SdkFilesystemTrimGetStatusRequest defines a request to get status of a background filesystem trim operation
+message SdkFilesystemTrimGetStatusRequest {
+  // Id of the volume
+  string volume_id = 1;
+  // Path where the volume is mounted
+  string mount_path = 2;
+}
+
+// SdkFilesystemTrimGetStatusResponse defines the response for a SdkFilesystemTrimGetStatusRequest. If case of error, this response struct is not valid
+message SdkFilesystemTrimGetStatusResponse{
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem trim GetStatus request
+  string status = 2;
+}
+
+// SdkFilesystemTrimStopRequest defines a request to stop a background filesystem trim operation
+message SdkFilesystemTrimStopRequest {
+  // Id of the volume
+  string volume_id = 1;
+  // Path where the volume is mounted
+  string mount_path = 2;
+}
+
+// SdkFilesystemTrimStopResponse defines the response for a SdkFilesystemTrimStopRequest. If case of error, this response struct is not valid
+message SdkFilesystemTrimStopResponse{
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem trim Stop request
+  string status = 2;
+}
+
+// SdkFilesystemCheckReportRequest defines a request to report the filesystem consistency issues
+message SdkFilesystemCheckReportRequest {
+  // Id of the volume
+  string volume_id = 1;
+}
+
+// SdkFilesystemCheckReportResponse defines the response for a SdkFilesystemCheckReportRequest. If case of error, this response struct is not valid
+message SdkFilesystemCheckReportResponse{
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem check Report request
+  string status = 2;
+}
+
+// SdkFilesystemCheckFixRequest defines a request to fix all the filesystem consistency issues reported in the response to OpenStorageFilesystemCheckService.Report() request.
+message SdkFilesystemCheckFixRequest {
+  // Id of the volume
+  string volume_id = 1;
+}
+
+// SdkFilesystemCheckFixResponse defines the response for a SdkFilesystemCheckFixRequest. If case of error, this response struct is not valid
+message SdkFilesystemCheckFixResponse{
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem check Fix request
+  string status = 2;
+}
+
+// SdkFilesystemCheckGetStatusRequest defines a request to get status of a background filesystem check operation
+message SdkFilesystemCheckGetStatusRequest {
+  // Id of the volume
+  string volume_id = 1;
+}
+
+// SdkFilesystemCheckGetStatusResponse defines the response for a SdkFilesystemCheckGetStatusRequest. If case of error, this response struct is not valid
+message SdkFilesystemCheckGetStatusResponse{
+  // Id of the volume
+  string volume_id = 1;
+  // Status of the background filesystem check GetStatus request
+  string status = 2;
```

Comment:
`status` should be some type of enum. Maybe add a `message` for a string returned from the code.

Reply:
enum displaying job status